mlflow/transformers.py

```diff
@@ -1903,6 +1906,29 @@ def _parse_json_encoded_dict_payload_to_dict(data, key_to_unpack):
                 }
                 for entry in data
             ]
+        elif isinstance(data, dict):
+            # This is to handle serving use cases as the DataFrame encapsulation converts
+            # collections within rows to np.array type. In order to process this data through
+            # the transformers.Pipeline API, we need to cast these arrays back to lists
+            # and replace the single quotes with double quotes after extracting the
+            # json-encoded `table` (a pandas DF) in order to convert it to a dict that
+            # the TableQuestionAnsweringPipeline can accept and cast to a Pandas DataFrame.
+            output = {}
+            for key, value in data.items():
+                if key == key_to_unpack:
+                    if isinstance(value, np.ndarray):
+                        output[key] = json.loads(str(value.tolist()).replace("'", '"'))
```

Comment:
relying on the __str__ of the list using single quotes seems pretty brittle / vulnerable to breakage. Also, what if the __str__ of the list gets truncated for long inputs? Then we'd blow uup.

 Instead, can you read the individual values and put double quotes around them?

Reply:
What happens if the values themselves contain single or double quotes? Have we tested that case / can we handle it & add tests?

Reply:
Yep, this was a broken fragile mess. Moved to ast.literal_eval and direct access to np.ndarray elements via `array.item()` so that any internal characters within the array values are preserved as-is and cast appropriately to python primitives.

Reply:
@BenWilson2 can you push changes (would love to see how this works with ast eval)